tests/maintel/test_mtcs.py

```diff
@@ -750,6 +752,451 @@ async def test_stop_all(self):
         with self.assertRaises(NotImplementedError):
             await self.mtcs.stop_all()
 
+    async def test_raise_m1m3(self):
+
+        self._mtm1m3_evt_detailed_state.detailedState = (
+            idl.enums.MTM1M3.DetailedState.PARKED
+        )
+
+        await self.mtcs.raise_m1m3()
+
+        self.mtcs.rem.mtm1m3.evt_detailedState.aget.assert_awaited()
+        self.mtcs.rem.mtm1m3.evt_detailedState.flush.assert_called()
+        if hasattr(self.mtcs.rem.mtm1m3.cmd_raiseM1M3.DataType(), "raiseM1M3"):
+            self.mtcs.rem.mtm1m3.cmd_raiseM1M3.set_start.assert_awaited_with(
+                raiseM1M3=True, timeout=self.mtcs.long_timeout
+            )
+        else:
+            self.mtcs.rem.mtm1m3.cmd_raiseM1M3.set_start.assert_awaited_with(
+                timeout=self.mtcs.long_timeout
+            )
+        self.mtcs.rem.mtm1m3.evt_detailedState.next.assert_awaited_with(
+            flush=False, timeout=self.mtcs.m1m3_raise_timeout
+        )
+
+    async def test_raise_m1m3_when_active(self):
+
+        self._mtm1m3_evt_detailed_state.detailedState = (
+            idl.enums.MTM1M3.DetailedState.ACTIVE
+        )
+        await self.mtcs.raise_m1m3()
+
+        self.mtcs.rem.mtm1m3.evt_detailedState.aget.assert_awaited()
+        self.mtcs.rem.mtm1m3.evt_detailedState.flush.assert_not_called()
+        self.mtcs.rem.mtm1m3.cmd_raiseM1M3.set_start.assert_not_awaited()
+        self.mtcs.rem.mtm1m3.evt_detailedState.next.assert_not_awaited()
+
+    async def test_raise_m1m3_aborted(self):
+
+        self._mtm1m3_evt_detailed_state.detailedState = (
+            idl.enums.MTM1M3.DetailedState.PARKED
+        )
+
+        m1m3_raise_task = asyncio.create_task(self.mtcs.raise_m1m3())
+
+        await asyncio.sleep(2.0)
```

Comment:
Is there a way to avoid explicitly sleeping here? Maybe wait for a task or something?

Reply:
I see you point. But what I want to do in this case is allow the raise to start, run for a bit and then abort it. So, there's really no intermediate task that I can wait for. What I do is wait long enough for the task to start executing but short enough such that it does not complete.